home/import_assessment_pages.py

```diff
@@ -0,0 +1,361 @@
+    question: list[str] | list[dict[str, Any]] | str = field(default_factory=list)
+    error: str = ""
+    answers: list[str] = field(default_factory=list)
+    scores: list[str] = field(default_factory=list)
+
+    @classmethod
+    def from_flat(cls, row: dict[str, str]) -> "ContentRow":
+        class_fields = {field.name for field in fields(cls)}
+        row = {
+            key.strip(): value.strip()
+            for key, value in row.items()
+            if value and key in class_fields
+        }
+        try:
+            return cls(
+                tags=deserialise_list(row.pop("tags", "")),
+                question=[row.pop("question")] if row.get("question") else "",
+                answers=deserialise_list(row.pop("answers", "")),
+                scores=deserialise_list(row.pop("scores", "")),
+                **row,  # type: ignore
+            )
+        except TypeError:
+            raise ImportAssessmentException(
+                "The import file is missing some required fields."
+            )
+
+
+def group_rows_by_page_id(rows: list[ContentRow]) -> list[ContentRow]:
+    grouped_rows = defaultdict(list)
+
+    for row in rows:
+        grouped_rows[row.slug].append(row)
+    output_rows = []
+    for _slug, grouped_rows_list in grouped_rows.items():
+        questions = []
+        for grouped_row in grouped_rows_list:
+            for _, question in enumerate(grouped_row.question):
+                answers = grouped_row.answers if grouped_row.answers else []
+                scores = grouped_row.scores if grouped_row.scores else []
+                error = grouped_row.error if grouped_row.error else []
+                question_entry = {
+                    "question": question,
+                    "answers": [
+                        {"answer": ans.strip(), "score": sc.strip()}
+                        for ans, sc in zip(answers, scores, strict=False)
+                    ],  # Strip whitespace
+                    "error": error,
+                }
+                questions.append(question_entry)
+
+        output_rows.append(
+            ContentRow(
+                slug=grouped_rows_list[0].slug,
+                title=grouped_rows_list[0].title,
+                tags=grouped_rows_list[0].tags,
+                locale=grouped_rows_list[0].locale,
+                high_result_page=grouped_rows_list[0].high_result_page,
+                high_inflection=grouped_rows_list[0].high_inflection,
+                medium_result_page=grouped_rows_list[0].medium_result_page,
+                medium_inflection=grouped_rows_list[0].medium_inflection,
+                low_result_page=grouped_rows_list[0].low_result_page,
+                low_inflection=grouped_rows_list[0].low_inflection,
+                generic_error=grouped_rows_list[0].generic_error,
+                question_count=str(len(questions)),
+                question=questions,
+                answers=[],
+                scores=[],
```

Comment:
Why are we putting the question, answers, and scores all into the `question` field, and then leaving `answers` and `scores` blank? 

I think the problem here is that we're trying to use `ContentRow`, which is a single row in the spreadsheet, to represent a `ShadowAssessmentPage`, which is made up of a collection of rows in a spreadsheet, one for each question. 

I think this function should rather be taking in `ContentRow`s, grouping those rows, and returning `ShadowAssessmentPage`s